docs/src/main/tut/add-support-for-complex-types.md

```diff
@@ -0,0 +1,246 @@
+### Add support for complex types
+
+Support for more complex types, such as Java configurations, can be achieved in many different methods,
+each one with pros and cons. While there is a preferable method, we will
```

Comment:
I'd be tempted to add a direct link to the preferred method. This paragraph conveys more of an air of mystery than developers are generally expecting when scanning documentation for how to get poop done.